tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/multiprotocol/JMSMessageConsumerTest.java

```diff
@@ -193,4 +195,66 @@ private void testDurableSubscriptionWithConfigurationManagedQueue(ConnectionSupp
          assertEquals("color = 'BLUE'", queue.getFilter().getFilterString().toString());
       }
    }
+
+   @Test(timeout = 30000)
+   public void testMapMessageConversionBetweenOpenWireAndAMQP() throws Exception {
+      testMapMessageConversion(createOpenWireConnection(), createConnection());
+   }
+
+   @Test(timeout = 30000)
+   public void testMapMessageConversionBetweenAMQPAndOpenWire() throws Exception {
+      testMapMessageConversion(createConnection(), createOpenWireConnection());
+   }
+
+   @Test(timeout = 30000)
+   public void testMapMessageConversionBetweenCoreAndAMQP() throws Exception {
+      testMapMessageConversion(createCoreConnection(), createConnection());
+   }
+
+   @Test(timeout = 30000)
+   public void testMapMessageConversionBetweenAMQPAndCore() throws Exception {
+      testMapMessageConversion(createConnection(), createCoreConnection());
+   }
+
+   @Test(timeout = 30000)
+   public void testMapMessageConversionBetweenCoreAndOpenWire() throws Exception {
+      testMapMessageConversion(createCoreConnection(), createOpenWireConnection());
+   }
+
+   @Test(timeout = 30000)
+   public void testMapMessageConversionBetweenOpenWireAndCore() throws Exception {
+      testMapMessageConversion(createOpenWireConnection(), createCoreConnection());
+   }
+
+   private void testMapMessageConversion(Connection sender, Connection consumer) throws Exception {
```

Comment:
Something like senderConnection and consumerConnection would seem more descriptive...you could then follow through with 'sender[Foo]' and 'consumer[Foo]' convention for related objects and wouldnt need to call the sole consumer "consumer2".

Reply:
That's fair.